[Team-03,FE]-✨ 필터 드롭다운 기능 및 이슈 상태 제어 기능 통합 구현

y-minion opened the pull request:
## 🧾 작업 요약

이번 PR은 메인 페이지에서의 이슈 필터링과 상태 변경 UX 전반을 개선하고,
상태 동기화 및 드롭다운 인터랙션의 완성도를 높이기 위한 기능 개발과 리팩토링을 포함합니다.

⸻

## ✅ 주요 작업 내역

### 1. 🔽 필터 드롭다운 기능 구현 (FilterBar)
- 쿼리 파라미터 기반으로 필터 상태를 초기화 (initFilter)
- 사용자가 드롭다운 항목 클릭 시:
- 기존 쿼리 기반 필터를 초기화 후
- 선택한 필터를 갱신 (setFilter)
- 업데이트된 상태를 기반으로 쿼리 URL 갱신 (applyQueryParams)
- 유저 식별값(userId) 기반 조건 처리: 내가 작성한 이슈, 나에게 할당된 이슈, 댓글 남긴 이슈 등
- useFilterStore.getState() 사용으로 최신 상태 동기 처리

---
### 2. ✅ 드롭다운 애니메이션 개선
- 메뉴가 위에서 아래로 부드럽게 등장하도록 CSS keyframes 애니메이션 적용
- DropdownMenuTemplate.styles.js에 slideDown 애니메이션 추가
- Overlay 위치 조정으로 드롭다운 메뉴가 TriggerButton 기준으로 정확히 위치되도록 수정

---

### 3. 🧠 deepEqualFast 함수 도입 및 테스트 완료
- useEffect 내부에서 불필요한 리렌더링 방지를 위한 깊은 비교 함수 작성
- JSON.stringify 한계를 보완하며, 객체가 바뀐 경우만 감지하도록 설계
- deepEqualFast.test.js 작성 및 Vitest 기반 테스트 코드 구성 완료

---

### 4. ✅ 이슈 상태 일괄 변경 기능 (StatusEditDropDown)
- 선택된 이슈의 ID 배열을 바탕으로 서버에 PATCH 요청
- 서버에서는 해당 ID 목록의 이슈들의 isOpen 값을 toggle 하도록 구성
- 성공 시 부모(IssueListPage)에게 onPatchSuccess() 콜백으로 GET 재요청 트리거

---

### 5. 📦 체크박스 기반 이슈 다중 선택 기능
- useCheckBoxStore 구현 및 도입
- 각 이슈의 개별 체크, 전체 체크/해제, 상태 일괄 초기화 등 지원
- SelectDisplayer 컴포넌트에서 현재 선택된 이슈 개수 렌더링 구현

---

### 6. 🐛 기타 버그 수정
- 라벨 호버 시 툴팁이 나타나지 않던 문제 해결
- 서버에서 받은 라벨과 라벨 스토어의 ID 불일치 문제를 정확히 매칭되도록 수정
- DropdownMenuTemplate 내 키 컴포넌트명 오타(Triiger → Trigger) 수정


⸻

## 🚀 테스트 확인 내역
	•	✅ deepEqualFast 함수에 대한 단위 테스트 (Vitest 통과)
	•	✅ 드롭다운 필터 → URL 반영 확인
	•	✅ 상태 토글 후 GET 요청 재실행 → UI 반영 확인
	•	✅ 이슈 선택 상태에서 체크박스 동작 검증

crongro commented on fe/src/base-ui/utils/IssueStatusSidebar/AddStatusToggle.jsx:
```diff
@@ -0,0 +1,167 @@
+  pageTypeContext = null,
+}) {
+  const [open, setOpen] = useState(false);
+  // const { toggleAssignee, toggleLabel, toggleMilestone } = useIssueDetailStore(
+  //   (state) => ({
+  //     toggleAssignee: state.toggleAssignee,
+  //     toggleLabel: state.toggleLabel,
+  //     toggleMilestone: state.toggleMilestone,
+  //   }),
+  //   shallow,
+  // );
```
안쓰는 코드의 주석은 항상 삭제하고 커밋해보세요.

crongro commented on fe/src/base-ui/utils/IssueStatusSidebar/AddStatusToggle.jsx:
```diff
@@ -0,0 +1,167 @@
+
+  function handleCloseOverlay(context) {
+    setOpen((prev) => !prev);
+    if (context === 'detail') {
```
context명이 detail 보다 더 구체적이고 상수라면 대문자를 사용해서 구분해봐요.

crongro commented on fe/src/stores/IssueDetailStore.js:
```diff
@@ -0,0 +1,66 @@
+      });
+    },
+
+    toggleLabel: (item) => {
+      set((state) => {
+        const labels = state.labels;
+        if (someHandler(labels, item.id)) {
+          state.labels = labels.filter(({ id }) => id !== item.id);
+        } else {
+          state.labels.push(item);
+        }
+      });
+    },
```
위에 toogleAssignee랑 상당히 비슷한데요?

crongro commented on fe/src/stores/IssueDetailStore.js:
```diff
@@ -0,0 +1,66 @@
+    milestone: {},
+    comments: [],
+    //클릭하는 순간 적용
+    toggleAssignee: (item) => {
```
화살표 함수의 특별한 장점을 활용하는것이 아니면, 화살표함수 대신 간단히 표현하세요.

toggleAssignee(item) {

crongro commented on fe/src/units/deepEqualFast/deepEqualFast.js:
```diff
@@ -0,0 +1,17 @@
+  if (aKeys.length !== bKeys.length) return false;
+
+  for (let key of aKeys) {
+    if (!b.hasOwnProperty(key)) return false;
```
Object 의 hasownproperty 함수는 왜 존재하는지, 왜 체크해야하닌지 한번 알아보세요.

crongro commented on fe/src/units/kanbanHeader/StatusEditDropDown.jsx:
```diff
@@ -0,0 +1,71 @@
+export default function StatusEditDropDown({ onPatchSuccess }) {
+  const accessToken = useAuthStore((state) => state.accessToken);
+  const checkBoxEntry = useCheckBoxStore((state) => state.checkBoxEntry);
+  const fetchType = '이슈 상태 조작';
+  const { response, fetchData } = useDataFetch({ fetchType });
+  const triggerButtonLabel = '상태수정';
+  const dropDownLabel = '상태 변경';
```
변수 선언도 비슷한 것들을 모아서 두면 보기 좋아요. 

여기서는 문자열 변수들을 모아서 같이 선언하면 좋죠.
그리고 이런 비슷한것을 객체로 묶어서 관리해도 좋고요.

crongro commented on fe/src/utils/common/someHandler.test.js:
```diff
@@ -0,0 +1,20 @@
+    ];
+    const target = 12;
+
+    expect(someHandler(a, target)).toBe(true);
```
잘했고요. 
someHandler(a, target) 를 expect안에 바로 넣기보다, 
테스트 대상임을 표현하기 위해 먼저 실행하고 결과를 변수로 저장해서 사용하면 더 좋죠.

crongro commented on fe/src/utils/mapToUnifiedId/mapToUnifiedId.test.js:
```diff
@@ -0,0 +1,23 @@
+    expect(result[0]).toHaveProperty('labelId', 1);
+  });
+
+  it('should map milestoneId to id', () => {
+    const input = [{ milestoneId: 10, name: 'v1.0' }];
+    const result = mapToUnifiedId(input);
+    expect(result[0]).toHaveProperty('id', 10);
+    expect(result[0]).toHaveProperty('milestoneId', 10);
+  });
```
it 안에 설명은 자세하게 하면서 좀 길어도 되고, 한글로만 해도 돼요.

crongro reviewed (approved):
테스트 코드까지 ~ 좋군요.
컴포넌트 테스트코드 많이 해보세요.

crongro commented:
@y-minion 앗 충돌이다..

crongro commented:
아직도 충돌이군요; @y-minion

dongchan0105 committed:
🚑️ RDS 주소 오타 수정

dongchan0105 committed:
🚑️database_이름 수정

dongchan0105 committed:
🚑️app.jar를 절대경로로 넣어서 dev 프로파일 찾도록 설정

dongchan0105 committed:
yml 파일 3개로 분리

Commit:
🐛 목서버에서 label 필드를 처리하지 못하던 버그 해결완료.

목서버에서 label의 id 를 labelId 라는 필드값으로 조회를 해야하지만, 기존에는 Id 로만 조회를 시도해서 이 부분을 수정 완료함.

dongchan0105 committed:
yml groups 지정

Commit:
✨ 댓글 등록 UI컴포넌트 추가

배포 위해 커밋하겠습니다.

jang-jinuk committed:
Merge pull request #114 from codesquad-masters2025-team03/fix/be/issue-detail-field-mismatch

[BE/MUD] - 이슈 상세 조회 API 응답 불일치한 필드명 수정 및 누락된 데이터 추가

jang-jinuk committed:
Merge pull request #112 from codesquad-masters2025-team03/fix/be/pagination-metadata-key-mismatch

[BE/MUD] - 페이지네이션 메타데이터 필드명 불일치 오류 해결

Commit:
Merge branch 'feat/fe/main' into dev

Commit:
♻️ 배포용 API 수정

수정했습니다.

dongchan0105 committed:
🐛 dev 프로파일을 읽을 수 있도록 설정

dongchan0105 committed:
Merge branch 'BE_REVIEW' into BE_Johnnie_newLogic_week3

dongchan0105 committed:
Merge pull request #115 from codesquad-masters2025-team03/BE_Johnnie_newLogic_week3

[BE Johnnie]- response Dto 상세내용들 재설정 및 rds 연결 CICD 완성

y-minion committed:
Merge branch 'dev' into BE_REVIEW

dongchan0105 committed:
Merge pull request #116 from codesquad-masters2025-team03/BE_REVIEW

Be review - 백엔드 코드 merge

dongchan0105 committed:
Merge branch 'main' into dev

dongchan0105 committed:
Merge pull request #117 from codesquad-masters2025-team03/dev

3주차 코드 통합

dongchan0105 committed:
Update deploy.yml [수정본]

Update deploy.yml [수정본]

dongchan0105 committed:
충돌 해결

dongchan0105 committed:
충돌 난 코드 해결

dongchan0105 committed:
♻️  OPTIONS 요청을 필터를 통과하도록 하고 CustomError 프론트에서 사용자에게 오류 원인 보여줄 수 있도록 설정

dongchan0105 committed:
OPTIONS 응답에 CORS 헤더를 직접 수동으로 추가

dongchan0105 committed:
CORS 헤더들 각각 넣도록 수정

dongchan0105 committed:
JwtAuthFilter에서 OPTIONS 요청은 완전히 우회시키기

dongchan0105 committed:
로그 찍어보기

dongchan0105 committed:
🔥  서버 하나로 합치는 deploy 코드 작성

dongchan0105 committed:
♻️ jar 파일명 명시해서 cp 실패 수정

dongchan0105 committed:
♻️ dist 경로 오류 수정 및 zip 포함 경로 정리

dongchan0105 committed:
♻️ 프론트 dist 경로 중첩 제거 및 zip 포함 경로 수정

dongchan0105 committed:
merge

dongchan0105 committed:
♻️ 프론트 dist 경로 중첩 제거 및 zip 포함 경로 수정

dongchan0105 committed:
zip 루트 구조 변경

dongchan0105 committed:
[FIX] zip 루트 구조 수정 (bundle 디렉토리 추가)

dongchan0105 committed:
경로 풀기

dongchan0105 committed:
압축 부분 수정

dongchan0105 committed:
appspec.yml수정

dongchan0105 committed:
yml ㅜ정

dongchan0105 committed:
충돌 해결

dongchan0105 committed:
[FIX] FRONTEND_API_BASE_URL 환경변수 연동

dongchan0105 committed:
수정요소들 변경

dongchan0105 committed:
GitHub Actions 워크플로우 전체 수정

dongchan0105 committed:
환경변수 주입

dongchan0105 committed:
폴더 구조 수정

dongchan0105 committed:
dd

dongchan0105 committed:
♻️ React가 사용하는 모든 리소스 요청이 JWT 필터예외 처리

dongchan0105 committed:
♻️ vite.svg 대응 추가

dongchan0105 committed:
로그 찍어보기

jang-jinuk committed:
✅ 이슈 상태 변경 요청 DTO 추가

- 이슈 ID 목록과 변경할 상태 값을 전달받는 DTO 추가

jang-jinuk committed:
✅ 다중 이슈의 is_open 상태를 일괄 변경하는 기능 추가

- JdbcTemplateIssueRepository에 updateIsOpenBulk 메서드 추가
- IN 절을 사용하여 여러 issue_id에 대해 is_open 값을 동시에 업데이트

dongchan0105 committed:
/app/frontend/ → Spring Boot /static/ 매핑 없이 자동 서빙

jang-jinuk committed:
✅ 이슈 상태 일괄 변경 API 추가

- PATCH /toggleStatus 엔드포인트 추가
- 요청 받은 이슈 ID 목록과 상태값을 기반으로 서비스 로직 호출

dongchan0105 committed:
📦 Zip Deployment Files

dongchan0105 committed:
-f 조건문으로 이미 복사된 파일이라면 다시 복사 안 함

dongchan0105 committed:
디렉토리 구조 변경

dongchan0105 committed:
진짜 너무 힘들어요

dongchan0105 committed:
복사 조건 수정

dongchan0105 committed:
안되면 배포 안함

jang-jinuk committed:
🧪 이슈 상태 일괄 변경 기능 테스트 작성

- issueService의 updateIssueOpenState가 issueRepository의 updateIsOpen을 잘 실행하는지 테스트 함
- 테스트를 위한 @AllArgsConstructor 추가

dongchan0105 committed:
강제로 밀어버리기

dongchan0105 committed:
이거 하면 쥐피티가 된데

dongchan0105 committed:
필터 로직 수정

dongchan0105 committed:
필터 수정

dongchan0105 committed:
yml 파일 재 설정

Commit:
✨ 메인 코멘트 컴포넌트 연결 완료

이슈 등록의 메인 코멘트 컴포넌트 페이지 안에 렌더링 되게 구현완료.
-> 메인 코멘트 편집 기능 구현 필요, 기존의 코멘트 컴포넌트 재사용하며 편집 기능 구현 예정

dongchan0105 committed:
새로운 ec2 생성

dongchan0105 committed:
deploy 코드 수정

dongchan0105 committed:
✨ 배포용 커밋

dongchan0105 committed:
✨ 배포용 커밋2

Commit:
✨ 댓글 content POST 요청 로직 구현 완료.

이슈 상세 페이지에서 해당하는 issueId에 새로운 comment 의 content 만을 POST 요청으로 새로운 데이터를 등록하도록 구현 완료.

dongchan0105 committed:
필터로직 변환

dongchan0105 committed:
Add application.env for GitHub Actions

Commit:
✨ FormData 바디 만들어 주는 함수 생성

데이터,파일을 입력하면 BE으로 보낼수 있는 formData 형식으로 변환해주는 함수 구현 완료

Commit:
♻️ NewIssueFooter컴포넌트 파일 POST로직 수정

파일 등록시 formData 형식으로 전송하도록 로직 수정 완료.

Commit:
✨ Express 에서 파일 응답 할 수 있도록 multer 라이브러리 적용

사용자가 파일 첨부 하면 처리 할 수 있도록 목서버에서 multer 라이브러리 적용

Commit:
♻️ 이슈 상세페이지의 새로운 코멘트 등록의 파일 업로드

새로운 커멘트 등록시 파일등록도 할 수 있도록 기능 추가 구현 완료.

Commit:
♻️ 기존 서브 코멘트 편집시 파일 등록 기능 추가

서브 코멘트에서 파일 업로드 기능 추가구현.

Commit:
♻️ 메인 코멘트 파일 첨부 기능 추가 완료

메인 코멘트의 파일 추가 기능 구현 완료.

dongchan0105 committed:
배포용 커밋

dongchan0105 committed:
✅ 정확하게 bootJar 결과만 복사

dongchan0105 committed:
또 수정

dongchan0105 committed:
s3 저장 로직 제거

dongchan0105 committed:
API 주소 초기화 및 DEPLOY 통합하여 배포

dongchan0105 committed:
이상한 자르 파일 들어오는거 수정

dongchan0105 committed:
Update deploy.yml [수정본]

Update deploy.yml [수정본]

dongchan0105 committed:
deploy 수정

dongchan0105 committed:
시크릿 키 추가

dongchan0105 committed:
최종 CICD

dongchan0105 committed:
환경변수 파일 만들어서 넣기

dongchan0105 committed:
환경변수 프론트 주입

dongchan0105 committed:
✅ start.sh 생성 (따옴표로 인자 감싸기)

dongchan0105 committed:
파일 나눠서 실행시키기

dongchan0105 committed:
스크립트 외부 관리 & 경로 문제 해결

dongchan0105 committed:
s3 업로드 권한 로컬에서 부여하는 로직 삭제

Commit:
✨ 이슈 id 필드값 관련

id -> issueId 로 수정되서 관련 로직 수정 완료.

Commit:
🐛 하드 코딩된 레이블,마일스톤 갯수 제거

실제로 서버로부터 받아온 레이블,마일스톤의 갯수를 파싱에 렌더링하도록 수정완료.

Commit:
✨ 마일스톤 스토어 구현

마일스톤 레이아웃 구현중 마일스톤 스토어 구현 완료.

Commit:
✨ 마일스톤의 목서버 응답 로직 추가

웹에서 GET 요청시 해당하는 마일스톤 관련 데이터 응답하도록 로직 추가

Commit:
✨ 마일스톤 입력폼 구현완료

마일스톤을 편집, 등록 할 수 있는 컴포넌트를 구현.

dongchan0105 committed:
♻️ Issue에서 update로직에서 모든 필드를 변경하지 않아도 에러가 생기지 않도록 변경

dongchan0105 committed:
♻️Jackson이 getter를 통해 JSON으로 변환 할 수 있도록 라벨,마일스톤 DTO 변경

Commit:
✨ 마일스톤 아이템

마일스톤 아이템 렌더링 구현 완료.

dongchan0105 committed:
♻️ 비즈니스로직들을 서비스 계층으로 다 이동시켰습니다 이 과정에서 예외처리도 서비스계층에서 처리하고 글로벌 오류핸들러를 통해 해결될 수 있도록 수정하였습니다.

jang-jinuk committed:
Merge pull request #119 from codesquad-masters2025-team03/feature/toggle-issue-status

[BE/MUD] - 다중 이슈 상태 변경 기능 구현

jang-jinuk committed:
✅ dev 브랜치 기준으로 application.yml 파일 동기화

jang-jinuk committed:
Merge branch 'BE_REVIEW' into feat/be/ouath-login

Commit:
✨ RefetchHandler 함수 도입

마일스톤 편집 작업 완료후 성공 응답 올떄마다 자동으로 다시 GET 요청 보내도록 useRef,useEffect사용해 로직 구현 완료

Commit:
✨ 배포 위한 마무리

jang-jinuk committed:
Merge pull request #111 from codesquad-masters2025-team03/feat/be/ouath-login

[BE/MUD] - Github OAuth2 로그인  기능 구현

Commit:
♻️ 배포 위해 API 수정

jang-jinuk committed:
Merge pull request #120 from codesquad-masters2025-team03/BE_REVIEW

[BE/MUD] Github OAuth 로그인 및 다중 이슈 상태 변경 기능 merge 요청

dongchan0105 committed:
🚀 OAuth관련 환경변수 새롭게 등록

y-minion committed:
Merge branch 'dev' into feat/fe/main

y-minion committed:
Merge pull request #121 from codesquad-masters2025-team03/feat/fe/main

[FE/Milo] - 배포 위한 dev 브랜치 머지 PR

Commit:
♻️ API 수정

dongchan0105 committed:
Merge branch 'BE_REVIEW' into be_Johnnie_newLogic_week4

dongchan0105 committed:
코드 Service로직에서 변화된것들 마저 수정

dongchan0105 committed:
Merge pull request #122 from codesquad-masters2025-team03/be_Johnnie_newLogic_week4

Be johnnie new logic week4[최종CICD 및 사소한 편집사항들 수정]

dongchan0105 committed:
Merge branch 'be_Johnnie_newLogic_week4' of https://github.com/codesquad-masters2025-team03/issue-tracker into be_Johnnie_newLogic_week4

dongchan0105 committed:
Merge pull request #123 from codesquad-masters2025-team03/be_Johnnie_newLogic_week4

Be johnnie new logic week4

dongchan0105 committed:
Merge pull request #124 from codesquad-masters2025-team03/BE_REVIEW

Week4 newLogic 과 Refactor 로직들 새롭게 첨부

Commit:
🐛 사용자 날짜 파싱 버그 해결

사용자가 endDate 입력시 올바르게 파싱해 서버에 합의된 날짜 포맷 형식으로 보낼수 있도록 수정 완료.

dongchan0105 committed:
✅  빠진 gradle 의존성 추가

dongchan0105 committed:
♻️JWT토근 우회경로 추가

dongchan0105 committed:
♻️API 수정

Commit:
✨ 마일 스톤 페이지 최종 구현 완료.

Commit:
Merge branch 'feat/fe/milestone' into feat/fe/main

dongchan0105 committed:
home api를 /home으로 변경

dongchan0105 committed:
메인페이지를 /home -> /으로 변경

Commit:
🐛 옵셔널 체이닝 방식으로 코드 수정

Commit:
Merge branch 'feat/fe/main' of github.com:codesquad-masters2025-team03/issue-tracker into feat/fe/main

Commit:
Merge branch 'feat/fe/label' into feat/fe/main

y-minion committed:
Merge pull request #125 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - 배포위한 PR요청

jang-jinuk committed:
✅ 이슈 목록 요청 url 변경

jang-jinuk committed:
✅ 코멘트 삭제 API 엔드포인트 추가

- DELETE /issues/{issueId}/comments/{commentId} 엔드포인트 구현
- commentService.deleteById 호출을 통해 댓글 삭제 처리
- 성공 시 BaseResponseDto 형태로 메시지 반환

jang-jinuk committed:
✅ dev 브랜치 기준으로 동기화

jang-jinuk committed:
Merge pull request #127 from codesquad-masters2025-team03/feat/be/delete-comments

[BE/MUD] - 코멘트 삭제 API 엔드포인트 구현

jang-jinuk committed:
✅ 이슈 ID로 모든 댓글 삭제 기능 추가

- CommentService에 deleteAllByIssueId(issueId) 메서드 추가
- CommentRepository 구현체에 deleteByIssueId(issueId) SQL 추가

jang-jinuk committed:
✅ 레이블 할당 해제 기능 구현

jang-jinuk committed:
✅ 담당자 할당 해제 기능 구현

jang-jinuk committed:
✅ 이슈 삭제 API 및 연관 데이터 정리 로직 추가

- /issues/{issueId} DELETE API 구현
- 이슈 삭제 시 댓글, 라벨, 담당자 연관 관계 정리 후 이슈 삭제
- 직접 SQL로 이슈 삭제하는 deleteById 메서드 구현

jang-jinuk committed:
Merge pull request #129 from codesquad-masters2025-team03/feat/be/delete-issue

[BE/MUD] - 이슈 삭제 기능 구현

jang-jinuk committed:
✅ 레이블 삭제 시 issue_label 관계 데이터 제거 기능 추가

레이블이 삭제될 경우, 해당 레이블이 연결된 이슈들과의 관계도 함께 제거되도록
IssueLabelRepository에 deleteByLabelId 메서드를 추가하고,
IssueLabelService에 이를 호출하는 deleteByLabelId 메서드를 구현함.

jang-jinuk committed:
✅ 레이블 삭제 API 엔드포인트 추가

jang-jinuk committed:
Merge pull request #131 from codesquad-masters2025-team03/feat/be/delete-label

[BE/MUD] - 라벨 삭제 기능 구현

jang-jinuk committed:
✅ 이슈의 마일스톤 제거 로직 추가

- 마일스톤 삭제 시 이슈의 마일스톤을 제거하는 로직 구현

jang-jinuk committed:
✅ 마일스톤 삭제 API 엔드포인트 추가

- 마일스톤을 삭제 시 이슈에 할당된 마일스톤을 제거하도록 구현

jang-jinuk committed:
Merge pull request #133 from codesquad-masters2025-team03/feat/be/delete-milestone

[BE/MUD] - 마일스톤 삭제 API 엔드포인트 추가

dongchan0105 committed:
♻️ batch 전략을 이용하여 기존에 쿼리를 여러번 나가는걸 단축

Commit:
✨ 라벨 페이지 컴포넌트 전체 구현 완료

목서버를 통해 에러없이 CRUD 가능함을 확인완료 했습니다.

Commit:
✨ 라벨 페이지 구현 1차 완료

삭제 모달은 Oauth 로그인 구현 완료 후, 구현 하겠습니다.

Commit:
Merge branch 'feat/fe/label' into feat/fe/main

Commit:
✨ OAuth 인증 관리 페이지 구현

BE 에서 토큰생성후 해당 페이지로 리다이렉트시 토큰을 로컬스토리지에 저장후, 메인페이지도 이동시키는 로직 구현완료.

Commit:
✨ OAuth 외부 링크 이동 로직 구현 완료.

GitHub OAuth URL 로 유저를 이동시켜 OAuth 로그인 하도록 함.

dongchan0105 committed:
데이터 양 줄이기

Commit:
♻️ 라우트 수정

토큰 발급 성공시 /oauth-success 라우트로 이동.

dongchan0105 committed:
Merge branch 'BE_REVIEW' into be_Johnnie_newLogic_week4

Commit:
Merge branch 'feat/fe/Label' into feat/fe/main

dongchan0105 committed:
Merge pull request #136 from codesquad-masters2025-team03/be_Johnnie_newLogic_week4

batch 전략을 이용하여 기존에 쿼리를 여러번 나가는걸 단축

dongchan0105 committed:
Merge pull request #137 from codesquad-masters2025-team03/BE_REVIEW

[BE/MUD&JOHNNIE] - 엔티티들 삭제하는 API 추가 및 쿼리 최적화

dongchan0105 committed:
[Refactor]  issues.js에 충돌마커 삭제

dongchan0105 committed:
Update data.sql-PK값을 수동으로 입력

dongchan0105 committed:
requestparam에 대한 default값 지정

Commit:
✨ 페이드 인 래퍼 컴포넌트 구현완료

새로고침시 자연스럽게 전페 페이지가 나타날수 있게 도와주는 페이드인 래퍼 컴포넌트 구현 완료

Commit:
✨ 페이드인 래퍼 컴포넌트 적용 완료

전체 레이아웃 컴포넌트에 페이드인 래퍼 컴포넌트로 감싸서 구현 완료.

dongchan0105 committed:
♻️ label의 칼럼 수정

jang-jinuk committed:
✅ 프론트 리다이렉트 url 정보 추가

- Github secret에 등록된 FRONTEND_API_BASE_URL를 불러와 환경변수에 저장하여 사용할 수 있도록 설정

jang-jinuk committed:
♻️ git login 기능 반환 형식 변경

- git login 후 로그인 성공 페이지로 토큰 담아 리다이렉션하도록 변경
- GithubLoginController 패키지 위치 이동

jang-jinuk committed:
Merge pull request #139 from codesquad-masters2025-team03/feat/oauth-redirect

[BE/MUD] - GitHub OAuth 로그인 후 프론트엔드 리디렉션 처리로 변경

Commit:
🐛 멀티 파트의 데이터 필드값 데이터 타입 지정

데이터 타입을 application/json으로 지정후 PATCH 요청

Commit:
✨ 테스트용 커밋

Commit:
Merge branch 'feat/fe/dev-base' into feat/fe/main

y-minion committed:
Merge pull request #141 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - 테스트용 머지 요청

Commit:
🐛 Total체크박스 버그 해결

기존에 total체크박스 클릭시 모두 선택되지 않던 버그 해결. id의 키 변수명이 바뀐 이슈였음.

jang-jinuk committed:
🐛 이슈목록 조회 시 담당자 이미지 로딩실패 오류 해결

- 누락된 담당자 이미지 url 필드 추가

Commit:
🐛 페이지 블락킹 기능 구현 완료

준비된 페이지가 끝나면 페이지 스와이프 블락킹 기능 추가완료

jang-jinuk committed:
🐛 sql syntax 오류 해결

- id -> label_id로 변경하여 오류 해결

jang-jinuk committed:
Merge pull request #142 from codesquad-masters2025-team03/fix/be/missing-profile-image-in-issue-list

[BE/MUD] - 이슈목록 조회 시 담당자 이미지 로딩실패 오류 해결

Commit:
🐛 필터 초기화 버튼 바로 반응 안되는 버그 수정

스토어의 초기화가 비동기적으로발생하기에 초기화된 쿼리파람을 바로 적용할 수 없었음. 디폴트 쿼리를 바로 적용하여 클릭하는 순간 초기화 되도록  수정

jang-jinuk committed:
🐛 issueIds가 빈 리스트일 때 SQL 문법 오류 방지

- issueIds가 빈 리스트일 경우 IN () 구문으로 인한 SQL 오류 발생 문제 해결
- 빈 리스트일 때는 쿼리를 실행하지 않고 빈 결과를 바로 반환하도록 처리

dongchan0105 committed:
♻️ left  조인에서 조인할 데이터가 없을경우 inner조인되는것을 처리 -IssueFilterQueryBuilder , countGroupedIssues에서 쿼리 개수를 줄 일 수 있는 방안들은 추가

jang-jinuk committed:
Merge pull request #143 from codesquad-masters2025-team03/fix/be/filter-bug

[BE/MUD] - 이슈 필터링 오류 해결

dongchan0105 committed:
🐛 Issue 최초등록시 마일스톤을 기입하지 않으면 nullPointer 예외가 발생하는걸 해결

Commit:
🐛 메인페이지의 상단 필터 FilterBar 컴포넌트 버그 수정

적용된 필터를 한번 더 누르면 해제할 수 있도록 기능 수정완료

dongchan0105 committed:
✅  마일스톤관련하여 마일스톤과 연결된 이슈들의 열린개수와 닫힌 개수들 조회하여 response에 빌드

Commit:
🐛 파일확장자 조건 로직 수정

함수실행하지 않고 조건비교를 하여 항상 truthy가 나오던 버그 발생.

jang-jinuk committed:
✅ API 반환 형식 일치화

- 모든 API에 반환형식을 BaseResponseDto를 통해 반환하도록 구현

jang-jinuk committed:
Merge pull request #144 from codesquad-masters2025-team03/feat/be/unify-response-format

[BE/MUD] API 반환 형식 일치화

Commit:
Merge branch 'feat/fe/dev-base' into feat/fe/main

dongchan0105 committed:
Merge branch 'BE_REVIEW' into be_Johnnie_newLogic_week4

dongchan0105 committed:
Merge pull request #146 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - 머지요청 PR

dongchan0105 committed:
Merge pull request #145 from codesquad-masters2025-team03/be_Johnnie_newLogic_week4

Be johnnie new logic week4

dongchan0105 committed:
Merge pull request #147 from codesquad-masters2025-team03/BE_REVIEW

Be review MUD&JOHNNIE

Commit:
🐛 첨부파일 분기처리 로직 수정

issueFileUrl이 null이거나 빈 값일 경우, <img>나 <a>가 렌더링되지 않도록 처리되었습니다. 사용자 입력이 없을 때는 아무것도 표시되지 않아 UI가 깔끔하게 유지됩니다.

jang-jinuk committed:
✅ 코멘트 관련 API 반환 형식 일치화

- 코멘트 관련 API의 반환형식을 BaseResponseDto를 통해 반환하도록 구현

jang-jinuk committed:
Merge pull request #148 from codesquad-masters2025-team03/feat/be/unify-response-format

[BE/MUD] - 코멘트 관련 API 반환 형식 일치화

jang-jinuk committed:
✅ JwtAuthFilter에서 /oauth-success 경로 허용

jang-jinuk committed:
Merge pull request #149 from codesquad-masters2025-team03/feat/be/allow-oauth-success-in-jwt-filter

[BE/MUD] - JwtFilter에 /oauth-success 경로 허용

Commit:
🐛 드랍 다운 선택 토글 기능 개선

같은 항목 선택시 선택 해제하도록 개선 완료

dongchan0105 committed:
♻️ DB 필드에 issueFileUrl이 없던거 수정

dongchan0105 committed:
Merge pull request #150 from codesquad-masters2025-team03/be_Johnnie_newLogic_week4

♻️ DB 필드에 issueFileUrl이 없던거 수정

dongchan0105 committed:
Merge pull request #151 from codesquad-masters2025-team03/BE_REVIEW

Final Refactor ver2 Johnnie & Mud

Commit:
✨ 이슈 삭제 로직 구현완료

구현 완료.

Commit:
🐛 멀티파트 바디 수정

PATCH 요청시 헤더의 타입 삭제함.

Commit:
Merge branch 'feat/fe/dev-base' into feat/fe/main

y-minion committed:
Merge pull request #152 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - PR 요청

Commit:
🐛 JWT 토큰 보정로직 추가

토큰 형식 보정 로직 추가

Commit:
Merge branch 'feat/fe/dev-base' into feat/fe/main

y-minion committed:
Merge pull request #153 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - 최종 머지 PR 요청

Commit:
🐛 긴급 버그 수정

등록하면 모달 닫히게 로직 수정

y-minion committed:
Merge pull request #154 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - 🐛 긴급 버그 수정

Commit:
🐛 버그 추가 수정

수정 완료

y-minion committed:
Merge pull request #156 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo]🐛 버그 추가 수정

Commit:
🐛 삭제시 메인페이지로 리다이렉트 이동 안되는 이슈

useNavigate() 로 초기화후 메인페이지로 리다이렉트

Commit:
🐛 타임존 커터 도입

백엔드와 시간양식  맞추기 위해 타임존 커터 함수 도입

Commit:
✨ 버그 추적

추적

y-minion committed:
Merge pull request #157 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - PR 요청

Commit:
🐛 마일스톤 편집시 넘기는 데이터 포맷 변경

완료함.

Commit:
🐛 벡엔드와의 데이터 포멧 싱크 동기화

완료했습니다

y-minion committed:
Merge pull request #158 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - 백엔드와 데이터 포맷 싱크 동기화 완료

Commit:
🐛 Zjalt

s

y-minion committed:
Merge pull request #159 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - PR요청

Commit:
🐛 IsOpen 필드가 빠진 버그 해결

필드값 패치 핸들러에 추가하여 백엔드와의 데이터 포맷 싱크 완료

Commit:
🐛 IsOpen 필드키 누락 버그

해결 시도

y-minion committed:
Merge pull request #160 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - PR 요청

Commit:
🐛 날짜 파싱 방법 변경

신뢰성 높은 정확한 파싱방법으로 전환

y-minion committed:
Merge pull request #161 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - 🐛 날짜 파싱 방법 변경

Commit:
🐛 시간대 설정 수정

y-minion committed:
Merge pull request #162 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - 🐛 시간대 설정 수정

jang-jinuk committed:
♻️ 필드명 변경 nickname -> nickName

Commit:
🐛 날짜 시스템 수정

직접 파싱후 직접 물자열 생성하는 방식으로 확실하게 데이터 형식 맞춰서 보냄.

jang-jinuk committed:
♻️ 필드명 변경

- assigneeIds -> assigneeId 변경
- labelIds -> labelId 변경

y-minion committed:
Merge pull request #163 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - 날짜시스템 수정

jang-jinuk committed:
Merge remote-tracking branch 'origin/dev' into dev

jang-jinuk committed:
♻️ getter 메소드명 변경

Commit:
🐛 날짜 시스템 수정

y-minion committed:
Merge pull request #164 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - 🐛 날짜 시스템 수정

Commit:
✨ 마일스톤 삭제 로직 구현

삭제 로직 구현 완료

jang-jinuk committed:
✅ SPA 라우팅을 위한 index.html fallback 설정 추가

- React Router의 BrowserRouter를 사용할 때 직접 URL 접근 시 404 에러가 발생하는 문제를 해결하기 위해,
Spring WebMvcConfigurer를 통해 파일 경로가 아닌 모든 요청을 index.html로 포워딩하도록 설정함.

Commit:
🐛 OAuth 로그인시 발급 받은 토큰 로컬스토리지에 추가로 저장

.

y-minion committed:
Merge pull request #165 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - OAuth 로그인 로직 수정

Commit:
✨ 편집 권한에 따른 접근성 기능 구현

.

y-minion committed:
Merge pull request #166 from codesquad-masters2025-team03/feat/fe/main

FE,Milo] - ✨ 편집 권한에 따른 접근성 기능 구현

Commit:
🐛 새로고침시 토큰을 첨부하지 않던 버그 수정

해결완료.

y-minion committed:
Merge pull request #167 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - 긴급수정-> 🐛 새로고침시 토큰을 첨부하지 않던 버그 수정

Commit:
🐛 라우터 구조 수정

.

y-minion committed:
Merge pull request #168 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - 🐛 라우터 구조 수정

Commit:
🐛 라우터 네비게이터 수정

.

y-minion committed:
Merge pull request #169 from codesquad-masters2025-team03/feat/fe/main

[FE,Milo] - 🐛 라우터 네비게이터 수정

dongchan0105 committed:
➕  데이터 개수 추가
